src/app/common/hide-balance-provider.tsx

```diff
@@ -0,0 +1,20 @@
+import { createContext, useContext } from 'react';
+
+interface HideBalanceContextProps {
+  hideBalance?: boolean;
+}
+const hideBalanceProvider = createContext<HideBalanceContextProps | null>(null);
+
+const { Provider } = hideBalanceProvider;
+
+export function useHideBalanceContext() {
+  return !!useContext(hideBalanceProvider)?.hideBalance;
+}
+
+interface HideBalanceProviderProps {
+  children: React.ReactNode;
+  hideBalance?: boolean;
+}
+export function HideBalanceProvider({ children, hideBalance }: HideBalanceProviderProps) {
+  return <Provider value={{ hideBalance }}>{children}</Provider>;
+}
```

Comment:
We use redux for global state so can this be managed there rather than adding a new context?